src/transformStateWithClones.js

```diff
@@ -7,7 +7,45 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = addProperties(currentState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        currentState = removeProperties(currentState, action.keysToRemove);
+        break;
+
+      case 'clear':
+        currentState = clearProperties();
+        break;
+    }
```

Comment:
[CODE STYLE]: A `default` case should be added to the `switch` statement for error handling. This ensures that if an action type is passed that does not match any case, the function can handle it appropriately, for example by throwing an error or ignoring the action.